app/backend/core/authentication.py

```diff
@@ -138,128 +135,46 @@ def get_token_auth_header(headers: dict) -> str:
-            "groups/any(g:search.in(g, '{}'))".format(", ".join(auth_claims.get("groups", [])))
-            if use_groups_security_filter
-            else None
-        )
-
-        # If only one security filter is specified, use that filter
-        # If both security filters are specified, combine them with "or" so only 1 security filter needs to pass
-        # If no security filters are specified, don't return any filter
-        security_filter = None
-        if oid_security_filter and not groups_security_filter:
-            security_filter = f"{oid_security_filter}"
-        elif groups_security_filter and not oid_security_filter:
-            security_filter = f"{groups_security_filter}"
-        elif oid_security_filter and groups_security_filter:
-            security_filter = f"({oid_security_filter} or {groups_security_filter})"
-
-        # If global documents are allowed, append the public global filter
-        if self.enable_global_documents:
-            global_documents_filter = "(not oids/any() and not groups/any())"
-            if security_filter:
-                security_filter = f"({security_filter} or {global_documents_filter})"
-
-        return security_filter
-
-    @staticmethod
-    async def list_groups(graph_resource_access_token: dict) -> list[str]:
-        headers = {"Authorization": "Bearer " + graph_resource_access_token["access_token"]}
-        groups = []
-        async with aiohttp.ClientSession(headers=headers) as session:
-            resp_json = None
-            resp_status = None
-            async with session.get(url="https://graph.microsoft.com/v1.0/me/transitiveMemberOf?$select=id") as resp:
-                resp_json = await resp.json()
-                resp_status = resp.status
-                if resp_status != 200:
-                    raise AuthError(error=json.dumps(resp_json), status_code=resp_status)
-
-            while resp_status == 200:
-                value = resp_json["value"]
-                for group in value:
-                    groups.append(group["id"])
-                next_link = resp_json.get("@odata.nextLink")
-                if next_link:
-                    async with session.get(url=next_link) as resp:
-                        resp_json = await resp.json()
-                        resp_status = resp.status
-                else:
-                    break
-            if resp_status != 200:
-                raise AuthError(error=json.dumps(resp_json), status_code=resp_status)
-
-        return groups
-
     async def get_auth_claims_if_enabled(self, headers: dict) -> dict[str, Any]:
         if not self.use_authentication:
             return {}
         try:
             # Read the authentication token from the authorization header and exchange it using the On Behalf Of Flow
-            # The scope is set to the Microsoft Graph API, which may need to be called for more authorization information
+            # The scope is set to Azure Search for authentication
```

Comment:
@mattgotteiner Is there an Azure AI Search documentation about the ACLs? Might be good place to link it